hypertrace-view-generator/hypertrace-view-generator/src/main/java/org/hypertrace/viewgenerator/generators/SpanEventViewGenerator.java

```diff
@@ -75,28 +84,32 @@ public Schema getSchema() {
   }
 
   private SpanEventView.Builder generateViewBuilder(
-      Event event,
-      ByteBuffer traceId,
-      Map<ByteBuffer, Event> eventMap,
-      Map<ByteBuffer, ByteBuffer> childToParentEventIds,
-      Map<ByteBuffer, Event> exitSpanToCalleeApiEntrySpanMap) {
+      final Event event,
+      final ByteBuffer traceId,
+      final Map<ByteBuffer, Event> eventMap,
+      final Map<ByteBuffer, ByteBuffer> childToParentEventIds,
+      final Map<ByteBuffer, Event> exitSpanToCalleeApiEntrySpanMap,
+      final Map<ByteBuffer, EventInfo> eventInfoMap) {
 
     SpanEventView.Builder builder = SpanEventView.newBuilder();
 
     builder.setTenantId(event.getCustomerId());
     builder.setSpanId(event.getEventId());
     builder.setEventName(event.getEventName());
-
-    // api_trace_id
-    ByteBuffer apiEntrySpanId =
-        EnrichedSpanUtils.getApiEntrySpanId(event, eventMap, childToParentEventIds);
-
-    builder.setApiTraceId(apiEntrySpanId);
-    if (event.getEventId().equals(apiEntrySpanId)) {
-      // set this count to 1 only if this span is the head of the Api Trace
-      builder.setApiTraceCount(1);
-    } else {
-      builder.setApiTraceCount(0);
+    builder.setApiTraceCount(0);
+    EventInfo eventInfo = eventInfoMap.get(event.getEventId());
+
+    if (eventInfo != null) {
+      builder.setTotalSpanCount(eventInfo.getTotalEventCount());
+      if (null != eventInfo.getApiTraceId()) {
+        ByteBuffer apiTraceId = eventInfo.getApiTraceId();
+        builder.setApiTraceId(apiTraceId);
+        // set this count to 1 only if this span is the head of the Api Trace
+        if (event.getEventId().equals(apiTraceId)) {
+          builder.setApiTraceCount(1);
```

Comment:
Nesting ifs and splitting the if and else cases (else here being line 99) make this significantly more confusing to read.